### ISSUE TITLE
ffs/manager: add auth token & unpublished api

### ISSUE THREAD

**Issue by jsign** [textileio/powergate#386]
Would be good to enable an _admin token_ to call _admin_ related APIs such as _Create_ or other kind of APIs. 

We're living on the premise of not being exposed to the public, but that shouldn't be blocked by just adding a simple auth here. Might be wise to separate the admin endpoint to an unpublished one for extra security.

**Comment by andrewxhill** [comment]
this one is starting to feel important in light of the up coming competition. @asutula @jsign not sure how hard it is though...

**Comment by jsign** [comment]
BTW, I can jump on this soon. Apart from creating an FFS instance, another candidate?
We can tune that set of APIs later anyway, just getting a bit ahead.

**Comment by jsign** [comment]
@andrewxhill , @asutula, maybe feels if an admin-token is specified, all APIs should be authenticated:
Maybe a rule like: if the API doesn't expect an FFS token, then it should receive the admin-token?

Even in read-only APIs, like reading indexing data, it might be a good idea to protect it with a token to avoid someone just simply flooding with requests?

**Comment by andrewxhill** [comment]
well… perhaps there is another way to dice it (3rd key?). because i can see a developer using ffs apis in a scope that doesn’t have access to the admin but wanting a straight-forward way to read some of the indices

**Comment by jsign** [comment]
@andrewxhill , so maybe:
- FFS token (for things that already is used)
- Admin token: for some APIs
- Developer token: for some other APIs

That would mean all APIs would be covered with some token.

**Comment by asutula** [comment]
Sorry need to read this more closely when I have time, but quick thought: there should be admin apis and then all other apis should be covered by what we now call the FFS token. My thinking is that in a future version of the Powergate API, the default mode of operation will be token-authenticated, scoped addresses and data etc... the word "FFS" will disappear from the API.

**Comment by asutula** [comment]
So basically, you're either an admin or you're a user with token access. That token gives you access to all non-admin apis, and those apis are scoped to your FFS instance where appropriate, but that is an internal implementation detail.

**Comment by jsign** [comment]
Yeah, that's another possibility. That would force to always create an FFS instance to interact with the API, compared to using a _developer token_ which doesn't need that. The only way I could think that is a problem is if the manager has the auto-funding feature enabled, and you just want to give a developer access to the APIs (and not really FFS).

In that case, you'd be funding a new wallet address just because you want an FFS token to be used in non-FFS APIs. But that could be solved if we enable some new opt in the `ffs.Create(...)` to possibly override the auto-funding mechanism, or similar.

@asutula , @andrewxhill , what do you think about this?  Convinces a bit more to only use admin-token in important APIs, and leave using a valid FFS token for the rest? (possibly circumventing auto-funding wallets with new opts in the future if that's a problem or similar).

**Comment by asutula** [comment]
We could also decouple token generation and storage from FFS creation. So, you create a token in order to interact with powergate. If you want/need to create an FFS instance, you call an API that does that and the FFS instance id is stored with a relationship to your powergate token.

**Comment by asutula** [comment]
FYI I've started on this in #659. So far, I've started to group admin API calls into an `AdminService` defined in `powergate_admin.proto`. Short term plan is to use a gRPC auth interceptor applied to all calls in that service. So far the service contains:
```
service PowergateAdminService {
  rpc CreateInstance(CreateInstanceRequest) returns (CreateInstanceResponse) {}
  rpc ListInstances(ListInstancesRequest) returns (ListInstancesResponse) {}
  rpc GetQueuedStorageJobs(GetQueuedStorageJobsRequest) returns (GetQueuedStorageJobsResponse) {}
  rpc GetExecutingStorageJobs(GetExecutingStorageJobsRequest) returns (GetExecutingStorageJobsResponse) {}
  rpc GetLatestFinalStorageJobs(GetLatestFinalStorageJobsRequest) returns (GetLatestFinalStorageJobsResponse) {}
  rpc GetLatestSuccessfulStorageJobs(GetLatestSuccessfulStorageJobsRequest) returns (GetLatestSuccessfulStorageJobsResponse) {}
}
```
(I moved `Create` and `ListAPIs` from the other FFS service into here)

**Comment by asutula** [comment]
All those job related calls allow you to query for job information across all FFS instances. You can filter by FFS instance id if you want and by data cids as well.

**Comment by jsign** [comment]
Nice, I have some half-baked work with an interceptor and added admin flags.
I can do some quick work and just leave it at the point of adding which GRPC prefixes are needed for the token check.
 
I wonder if might deserve to merge this work already in your branch?  I can do that if make sense to you.

**Comment by asutula** [comment]
Yea that would be great. We're definitely converging there, don't want to duplicate anything you've already done.

**Comment by jsign** [comment]
Nice, I'll finish some work and merge in your branch. We'll coordinate a bit closer to the _merging_ event.
